### DISCUSSION TITLE
Micrometer + Exemplars: should these work out-of-the-box?

### DISCUSSION THREAD

**Discussion by brunobastosg** [quarkusio/quarkus#31771] [Q&A]
I'm using Quarkus 2.16.4 and trying to send Exemplar data to Prometheus.

I added the extensions `quarkus-opentelemetry` and `quarkus-micrometer-registry-prometheus`.

I also started Prometheus with the flag `--enable-feature=exemplar-storage`.

I make some requests to my app, but when I call `/q/metrics`, the exemplar data is not showing. The traces are being generated though, as I can see them on log messages and  in Jaeger or Tempo.

Was that supposed to work out-of-the-box? Or is some extra configuration needed? I checked the documentation but couldn't find anything.

[a comment by an automated account is not shown]

**Comment by brunobat** [comment]
Exemplars support were added to Micrometer quarkus extension on this PR: https://github.com/quarkusio/quarkus/pull/28360

**Reply by brunobastosg** [reply]
Thanks for the reply. I was following that PR. And on the PR it has a 2.16.0.CR1 milestone. So I'm assuming it should be working in 2.16.4.Final.

My question is if it should work just by adding extensions `quarkus-opentelemetry` and `quarkus-micrometer-registry-prometheus`. Or does it need some extra config?

**Reply by brunobat** [reply]
It should, `quarkus-micrometer` is included in the prometheus registry

**Reply by brunobastosg** [reply]
I uploaded the project to GitHub at https://github.com/brunobastosg/quarkus-exemplars. I would appreciate it if you could take a look at it when you have the time. No need to rush, I'm just trying out the new features.

**Reply by rodrahul** [reply]
Hey @brunobastosg  any luck getting exemplars working with Quarkus? 
I'm having similar issues. I've a sample spring app and a quarkus app, both sending metrics and traces to the same otel collector. I do see exemplars in Prometheus when querying metrics sent by Spring app but nothing from the Quarkus app

**Reply by brunobastosg** [reply]
No, I couldn't get it working back then so I moved on to other things. When I have the time I'll try again with Quarkus' latest version.
